code/datums/antagonists/wizard.dm

```diff
@@ -0,0 +1,250 @@
+	var/list/objectives = list() //this should be base datum antag proc and list, todo make lazy
+	var/move_to_lair = TRUE
+
+/datum/antagonist/wizard/on_gain()
+	register()
+	if(give_objectives)
+		create_objectives()
+	equip_wizard()
+	if(move_to_lair)
+		send_to_lair()
+	. = ..()
+	if(allow_rename)
+		rename_wizard()
+
+/datum/antagonist/wizard/proc/register()
+	SSticker.mode.wizards |= owner
+
+/datum/antagonist/wizard/proc/unregister()
+	SSticker.mode.wizards -= src
+
+/datum/objective_team/wizard
+	name = "wizard team"
+
+/datum/antagonist/wizard/proc/create_wiz_team()
+	wiz_team = new(owner)
+	wiz_team.name = "[owner.current.real_name] team"
+	update_wiz_icons_added(owner.current)
+
+/datum/antagonist/wizard/proc/send_to_lair()
+	if(!owner || !owner.current)
+		return
+	if(!GLOB.wizardstart.len)
+		SSjob.SendToLateJoin(owner.current)
+		to_chat(owner, "HOT INSERTION, GO GO GO")
+	owner.current.forceMove(pick(GLOB.wizardstart))
+
+/datum/antagonist/wizard/proc/create_objectives()
+	switch(rand(1,100))
+		if(1 to 30)
+			var/datum/objective/assassinate/kill_objective = new
+			kill_objective.owner = owner
+			kill_objective.find_target()
+			objectives += kill_objective
+
+			if (!(locate(/datum/objective/escape) in owner.objectives))
+				var/datum/objective/escape/escape_objective = new
+				escape_objective.owner = owner
+				objectives += escape_objective
+		
+		if(31 to 60)
+			var/datum/objective/steal/steal_objective = new
+			steal_objective.owner = owner
+			steal_objective.find_target()
+			objectives += steal_objective
+
+			if (!(locate(/datum/objective/escape) in owner.objectives))
+				var/datum/objective/escape/escape_objective = new
+				escape_objective.owner = owner
+				objectives += escape_objective
+
+		if(61 to 85)
+			var/datum/objective/assassinate/kill_objective = new
+			kill_objective.owner = owner
+			kill_objective.find_target()
+			objectives += kill_objective
+
+			var/datum/objective/steal/steal_objective = new
+			steal_objective.owner = owner
+			steal_objective.find_target()
+			objectives += steal_objective
+
+			if (!(locate(/datum/objective/survive) in owner.objectives))
+				var/datum/objective/survive/survive_objective = new
+				survive_objective.owner = owner
+				objectives += survive_objective
+
+		else
+			if (!(locate(/datum/objective/hijack) in owner.objectives))
+				var/datum/objective/hijack/hijack_objective = new
+				hijack_objective.owner = owner
+				objectives += hijack_objective	
+	
+	for(var/datum/objective/O in objectives)
+		owner.objectives += O
+
+/datum/antagonist/wizard/on_removal()
+	unregister()
+	owner.RemoveAllSpells() // TODO keep track which spells are wizard spells which innate stuff
+	return ..()
+
+/datum/antagonist/wizard/proc/equip_wizard()
+	if(!owner)
+		return
+	var/mob/living/carbon/human/H = owner.current
+	if(!istype(H))
+		return
+	if(strip)
+		to_chat(world,"TODO: delete equipment")
```

Comment:
Assuming this is still WIP based on these

Reply:
Dependent on #31411, i'll plug it in in a moment,